### PR TITLE
[3199] - Removed default shipping method auto-selection on Shipping step

### PR DESCRIPTION
**Pull request by leonazava**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/3199

**Problem:**
* On the Shipping step, a delivery option was auto-selected by default
* In-store pickup delivery option has special logic that's supposed to render when a user selects it, but autoselecting was causing bugs for the method.
* Radio buttons didn't rerender as unchecked when they should've had
* If user selected an option other than In-Store Pickup, then clicked "Pick In-Store" in the top menu if they have already selected a shop location before, the delivery method wouldn't change in the state

**In this PR:**
* Removed the auto-selection of a delivery option on mount
* Modified the "Proceed to billing" button to be disabled if a user hadn't picked any delivery options
* Fixed a bug where radio buttons were changing types from uncontrolled to controlled
* Made the app change selectedShippingMethod to In-Store delivery when isPickInStoreMethodSelected state prop changes

### CODE REVIEW

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js [hunk 1]
```diff
@@ -105,27 +106,17 @@ export class CheckoutShippingContainer extends PureComponent {
         };
     }
 
-    componentDidUpdate(prevProps) {
-        const { shippingMethods: prevShippingMethods } = prevProps;
-        const { shippingMethods } = this.props;
+    componentDidUpdate() {
+        const { isPickInStoreMethodSelected, shippingMethods } = this.props;
+        const pickupMethod = shippingMethods.find((el) => el.carrier_code === 'instore');
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js:111]: Please move `instore` to the constant and put it to the `CheckoutShipping.config.js` file.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js (hunk 1 above)
Comment (leonazava) [reply packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js]: Replaced it with `STORE_IN_PICK_UP_ATTRIBUTE_CODE` from `StoreInPickUp.config`

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
Please, add the same logic as in Luma - Remove the checkmark from In-Store delivery if user selected it and store but then returned to shipping tab
https://drive.google.com/file/d/1MXgRsvP8ddmdYKXp-CU-hvh0sq4OtMXm/view?usp=sharing

**Review by zans-laksa — APPROVED** [review]

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.component.js [hunk 2]
```diff
@@ -103,14 +103,15 @@ export class CheckoutShipping extends PureComponent {
 
     renderActions() {
         const { selectedShippingMethod } = this.props;
+        const isDisabled = !selectedShippingMethod.carrier_code;
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.component.js:106]: What if the selected shipping method is undefined?

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js [hunk 3]
```diff
@@ -106,26 +107,19 @@ export class CheckoutShippingContainer extends PureComponent {
...
 
-        if (prevShippingMethods !== shippingMethods) {
-            this.resetShippingMethod();
-        }
-    }
-
-    resetShippingMethod() {
-        const { selectedShippingMethod: { method_code: selectedMethodCode = '' } } = this.state;
-        const { shippingMethods } = this.props;
+        const instore = shippingMethods.find((el) => el.method_code === 'pickup');
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js:113]: Please put strings to constants and store them in the *.conf.js files.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js (hunk 3 above)
Comment (leonazava) [reply packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js]: done

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**